src/x11/xf86Wacom.c

```diff
@@ -401,9 +401,11 @@ convertAxes(const WacomAxisData *axes, int *first_out, int *num_out, int valuato
 		case WACOM_AXIS_STRIP_Y: pos = 4; break;
 		case WACOM_AXIS_ROTATION: pos = 3; break;
 		case WACOM_AXIS_THROTTLE: pos = 4; break;
-		case WACOM_AXIS_WHEEL: pos = 5; break;
-		case WACOM_AXIS_RING: pos = 5; break;
-		case WACOM_AXIS_RING2: pos = 6; break;
+		case WACOM_AXIS_SCROLL_X: pos = 5; break;
+		case WACOM_AXIS_SCROLL_Y: pos = 6; break;
+		case WACOM_AXIS_WHEEL: pos = 7; break;
+		case WACOM_AXIS_RING: pos = 8; break;
+		case WACOM_AXIS_RING2: pos = 8; break;
```

Comment:
I've only skimmed this but: those axis positions are out of sync with the comment (e.g. `ninth valuator` is scroll y but here it's the fifth).

And given the wacom driver's age and somewhat quirky behaviours it's safest not to touch the existing valuators and just append yours.

**edit:** ok, this is fixed in the follow-up commit but my comment above about not touching the existing ones applies.

Reply:
I did that because the last valuator (abswheel2) only gets initialized if it is available and thus has to be last, if we want to keep statically defined valuator numbers and not define them at runtime, abswheel2 has to be the last one.